### PR TITLE
[BottomNavigation] Create Bottom Navigation Swift example

### PR DESCRIPTION
**Pull request by jgunaratne**
Create Bottom Navigation Swift example

Depends on: https://github.com/material-components/material-components-ios/pull/2088

### CODE REVIEW

**Comment by ianegordon** [comment]
I'll leave it to @willlarche as our resident Swift and Storyboard enthusiast.

**Review by willlarche — COMMENTED** [review]
So enthused to see this.

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    view.addSubview(bottomNavBar)
+
+    // Always show bottom navigation bar item titles.
+    bottomNavBar.titleVisibility = .always
+
+    // Cluster and center the bottom navigation bar items.
+    bottomNavBar.alignment = .centered
+
+    // Add items to the bottom navigation bar.
+    let tabBarItem1 = UITabBarItem(title: "Home", image: UIImage.init(named: "Home"), tag: 0)
```
Comment (willlarche) [inline components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: You can remove the `init` and just have `UIImage(name: "Home")`.

Try doing that for all `init`s

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      UITabBarItem(title: "Messages", image: UIImage.init(named: "Email"), tag: 0)
+    let tabBarItem3 =
+      UITabBarItem(title: "Favorites", image: UIImage.init(named: "Favorite"), tag: 0)
+    bottomNavBar.items = [ tabBarItem1, tabBarItem2, tabBarItem3 ]
+  }
+
+  override func viewWillLayoutSubviews() {
+    let size = bottomNavBar.sizeThatFits(view.bounds.size)
+    let bottomNavBarFrame = CGRect(x: 0,
+                                   y: self.view.bounds.size.height - size.height,
```
Comment (willlarche) [inline components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: You don't need `size` to get the height and width of a rect. Just do `view.bounds.height`. Do for all.

Also, remove every `self` and only add it back in if Xcode forces you (there are only a couple cases where you have to have it).

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  let bottomNavBar = MDCBottomNavigationBar()
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+    self.title = "Bottom Navigation (Swift)"
+
+    self.addChildViewController(appBar.headerViewController)
+    let color = UIColor(white: 0.2, alpha:1)
+    appBar.headerViewController.headerView.backgroundColor = color
+    appBar.navigationBar.tintColor = UIColor.white
```
Comment (willlarche) [inline components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Remove `UIColor` and just have `.white`. You can try this for all. There may be one case where it requires it.

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    appBar.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName : UIColor.white]
+    
+    commonBottomNavigationTypicalUseSwiftExampleInit()
+  }
+
+  required init?(coder aDecoder: NSCoder) {
```
Comment (willlarche) [inline components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Mark this as unavailable:

```
  @available(*, unavailable)
  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }
```

**Review by willlarche — DISMISSED** [review]

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift (hunk 3 above)
Comment (jgunaratne) [reply components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Done.

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift (hunk 4 above)
Comment (jgunaratne) [reply components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Done.

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift (hunk 1 above)
Comment (jgunaratne) [reply components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Done.

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift (hunk 2 above)
Comment (jgunaratne) [reply components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift]: Done.

File: components/BottomNavigation/src/MDCBottomNavigationBar.m [hunk 5]
```diff
@@ -326,8 +326,10 @@ - (void)setItems:(NSArray<UITabBarItem *> *)items {
       itemView.badgeValue = item.badgeValue;
     }
 #if defined(__IPHONE_10_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0)
-    if (item.badgeColor) {
-      itemView.badgeColor = item.badgeColor;
+    if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 10.0) {
```
Comment (willlarche) [inline components/BottomNavigation/src/MDCBottomNavigationBar.m:329]: We have to do all 3 of these conditions?

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  init() {
+    super.init(nibName: nil, bundle: nil)
+    self.title = "Bottom Navigation (Swift)"
+
+    self.addChildViewController(appBar.headerViewController)
+    let color = UIColor(white: 0.2, alpha:1)
+    appBar.headerViewController.headerView.backgroundColor = color
+    appBar.navigationBar.tintColor = .white
+    appBar.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName : UIColor.white]
```
Comment (willlarche) [inline components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift:35]: See if you can drop `UIColor`.

**Review by willlarche — APPROVED** [review]

File: components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift (hunk 6 above)
Comment (jgunaratne) [reply components/BottomNavigation/examples/BottomNavigationTypicalUseExample.swift:35]: It's needed here to compile.

File: components/BottomNavigation/src/MDCBottomNavigationBar.m (hunk 5 above)
Comment (jgunaratne) [reply components/BottomNavigation/src/MDCBottomNavigationBar.m:329]: #if is for codecov/Travis. UIDevice is for iOS 8, that doesn't seem to accept @available.

**Pull request merged**